lib/player.js

```diff
@@ -2002,7 +2002,28 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       // TODO: File a bug or send a PR to the compiler externs to fix this.
       const textTracks = /** @type {EventTarget} */(this.video_.textTracks);
       this.eventManager_.listen(
-          textTracks, 'addtrack', () => this.onTracksChanged_());
+          textTracks, 'addtrack', (e) => {
+            this.onTracksChanged_();
+            const track = e.track;
```

Comment:
Please move this entire piece of logic to a separate method and add comments on what each bit does throughout.

Reply:
I changed it to his own method and added various comments.